### PR TITLE
Update Connected Sites modal design

### PR DESCRIPTION
**Pull request by whymarrh**
This PR implements the new design for the _Connected Sites_ modal.

**Screenshots:**

**Design notes:**

- It does handle the case where the site name is too long
- It does handle the case where the domain name is too long
- It does handle the case where the list is too long and the manual add link is at the bottom

**Deviations from the designs:**

1. The "Go to site.com" link in the options popup has been dropped
2. As a consequence of (1), the options popup has also been dropped since it has but one item and the three dots icon has been replaced with the first option, the trash icon

---

Refs #8229, #8242, #8248, #8253, #8254

### CODE REVIEW

**Review by rickycodes — DISMISSED** [review]
LGTM!

image

**Comment by whymarrh** [comment]
@rickycodes slight change to the title case for the tests

**Comment by whymarrh** [comment]
@rickycodes also I fixed that bottom border from your screenshot

**Comment by rickycodes** [comment]
> @rickycodes also I fixed that bottom border from your screenshot

@whymarrh I just did a pull @ `d362c9c34` and they still look off?

image

**Comment by whymarrh** [comment]
> I fixed that **bottom** border

The bottom border was missing in your screenshot, a regression (between the time I took my screenshots in the OP and you took yours). I fixed that.

The margin on the right is for the scrollbar, that's from the popover.

**Comment by rickycodes** [comment]
Oh, I see there was one missing from the bottom. Hmm, I wonder if we should be concerned about the gap on the right? Even still I guess we can open a new PR to adjust that.

**Review by rickycodes — DISMISSED** [review]

File: ui/app/components/ui/popover/popover.component.js [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+        <div className="popover-header__title">
+          <h2 title={title}>
+            {
+              onBack
+                ? <i className="fas fa-chevron-left" onClick={onBack} />
+                : null
+            }
+            {title}
+          </h2>
+          <i className="fas fa-times" onClick={onClose} />
```
Comment (Gudahtt) [inline ui/app/components/ui/popover/popover.component.js]: This can no longer be focused, now that it's an `i` instead of a `button`

Comment (Gudahtt) [reply ui/app/components/ui/popover/popover.component.js]: The same goes for the back button as well

Comment (whymarrh) [reply ui/app/components/ui/popover/popover.component.js]: Updated, and now with titles as well

File: ui/app/components/app/connected-sites-list/connected-sites-list.container.js [hunk 2]
```diff
@@ -1,52 +1,16 @@
...
-    showDisconnectAccountModal: (domainKey, domain) => {
-      dispatch(showModal({ name: 'DISCONNECT_ACCOUNT', domainKey, domain }))
-    },
-    showDisconnectAllModal: () => {
-      dispatch(showModal({ name: 'DISCONNECT_ALL' }))
-    },
-    legacyExposeAccounts: (origin, account) => {
-      dispatch(legacyExposeAccounts(origin, [account]))
-    },
     getOpenMetamaskTabsIds: () => dispatch(getOpenMetamaskTabsIds()),
```
Comment (Gudahtt) [inline ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: I'm not sure this belongs here anymore - it seems that the state this triggers a refresh of is now used exclusively in `ui/apps/pages/connected-sites/connected-sites.container.js`

Comment (whymarrh) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: Moved around in 4991010

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]

**Review by rickycodes — APPROVED** [review]

**Pull request merged**